### PR TITLE
Describe new architecture

### PR DESCRIPTION
**Pull request by datenreisender**
This implements [NCP-2832](https://projecttools.nordicsemi.no/jira/browse/NCP-2832).

Add descriptions about the new architecture and also about how to develop the common code in `pc-nrfconnect-shared`. Some pages are just completely only about the old architecture, so for now I just added a warning.

### CODE REVIEW

File: core_development.md [hunk 1]
```diff
@@ -73,18 +76,62 @@ have a binary installation of nRF Connect for Desktop or
 [develop local apps](./app_development), the same apps will show up in this
 instance too.
 
-### Testing
+## Developing common code in [pc-nrfconnect-shared](https://github.com/NordicSemiconductor/pc-nrfconnect-shared)
+
+When you are developing common code in `pc-nrfconnect-shared` and you want to
+check the changes quickly in the launcher or an app the steps above are not
+sufficient, because by default the launcher will include a released version of
+`pc-nrfconnect-shared` from GitHub, not you local one. But with some additional
```
Comment (bencefr) [inline core_development.md:84]: typo: s/you/your/

File: core_development.md [hunk 2]
```diff
@@ -73,18 +76,62 @@ have a binary installation of nRF Connect for Desktop or
...
+1. Have both, `pc-nrfconnect-launcher` and `pc-nrfconnect-shared` checked out
+   into directories next to each other.
+2. In the directory `pc-nrfconnect-launcher` run
+```
+npm install; npm link ../pc-nrfconnect-shared
+```
+3. In the directory `pc-nrfconnect-shared` run
+```
+npm ci --prod
+```
```
Comment (bencefr) [inline core_development.md:97]: Very good to know!

File: getting_started.md [hunk 3]
```diff
@@ -39,31 +39,55 @@ exist:
...
   hardware
 
-A bit unusual: The common code is not only provided during development, the
-libraries are also provided during runtime, so that the individual apps do not
-have to provide them themselves.
+Where the last part, the common code, resides changes with the new architecture:
+Now it is found in project `pc-nrfconnect-shared` (since release
+[v4.8.0](https://github.com/NordicSemiconductor/pc-nrfconnect-shared/releases/tag/v4.8.0)
+of that project) but previously it was also included in
+``pc-nrfconnect-launcher`, where now only legacy variants remain for apps that
```
Comment (bencefr) [inline getting_started.md:70]: Here's a double back-tick that breaks markdown.

File: core_development.md (hunk 1 above)
Comment (datenreisender) [reply core_development.md]: Fixed in b25f665.

File: core_development.md (hunk 2 above)
Comment (datenreisender) [reply core_development.md:97]: I promised you that I will still describe it. ;-)

File: getting_started.md (hunk 3 above)
Comment (datenreisender) [reply getting_started.md]: Fixed in b25f665.

**Pull request merged**